### PR TITLE
Proposed API for property name casing using different enums for read and write

### PR DESCRIPTION
**Pull request by annelo-msft**
For discussion - one approach to https://github.com/Azure/azure-sdk-for-net/issues/36642

### CODE REVIEW

File: sdk/core/Azure.Core/api/Azure.Core.net461.cs [hunk 1]
```diff
@@ -17,9 +17,7 @@ public abstract partial class AsyncPageable<T> : System.Collections.Generic.IAsy
     }
     public static partial class AzureCoreExtensions
     {
-        public static dynamic ToDynamicFromJson(this System.BinaryData utf8Json) { throw null; }
-        public static dynamic ToDynamicFromJson(this System.BinaryData utf8Json, Azure.Core.Dynamic.DynamicCaseMapping caseMapping, Azure.Core.Dynamic.DynamicDateTimeHandling dateTimeHandling = Azure.Core.Dynamic.DynamicDateTimeHandling.Rfc3339) { throw null; }
-        public static dynamic ToDynamicFromJson(this System.BinaryData utf8Json, Azure.Core.Dynamic.DynamicDataOptions options) { throw null; }
+        public static dynamic ToDynamicFromJson(this System.BinaryData utf8Json, Azure.Core.Dynamic.PropertyNameLookup existingPropertyLookup, Azure.Core.Serialization.PropertyNameConversion newPropertyConversion = Azure.Core.Serialization.PropertyNameConversion.None, Azure.Core.Dynamic.DynamicDateTimeHandling dateTimeHandling = Azure.Core.Dynamic.DynamicDateTimeHandling.Rfc3339) { throw null; }
```
Comment (KrzysztofCwalina) [inline sdk/core/Azure.Core/api/Azure.Core.net461.cs:20]: I don't think we need to call the first parameter "existingPropertyLookup". "propertyLookup" would be clear enough.

Comment (KrzysztofCwalina) [reply sdk/core/Azure.Core/api/Azure.Core.net461.cs:20]: I thought we won't be using an enums for the DateTime  format

Comment (KrzysztofCwalina) [reply sdk/core/Azure.Core/api/Azure.Core.net461.cs:20]: BTW, another implication of this design is that users who write and want a mapping, will forever have to specify arguments. i.e. even if we add "CaseInsensitive" default to the lookup enum, the user who wants recasing on write will have to do:
```
dynamic json = data.ToDynamicFromJson(PropertyNameLookup.CaseInsensitive, PropertyNameConversion.CamelCase);
```

Comment (annelo-msft) [reply sdk/core/Azure.Core/api/Azure.Core.net461.cs:20]: > I don't think we need to call the first parameter "existingPropertyLookup". "propertyLookup" would be clear enough.

I'm happy to make that change.

Comment (annelo-msft) [reply sdk/core/Azure.Core/api/Azure.Core.net461.cs:20]: > I thought we won't be using an enums for the DateTime format

I'm tracking the work to rewrite that API as a separate feature.  I do plan to address it in a separate PR.

Comment (annelo-msft) [reply sdk/core/Azure.Core/api/Azure.Core.net461.cs:20]: > BTW, another implication of this design is that users who write and want a mapping, will forever have to specify arguments. i.e. even if we add "CaseInsensitive" default to the lookup enum, the user who wants recasing on write will have to do:

Why not?  Wouldn't it be possible to add an overload that takes PropertyNameConversion as the first parameter, or are there design guidelines against doing that?

Comment (KrzysztofCwalina) [reply sdk/core/Azure.Core/api/Azure.Core.net461.cs:20]: Yes, there are design guidelines stating that overloads should not reorder parameters.

Comment (annelo-msft) [reply sdk/core/Azure.Core/api/Azure.Core.net461.cs:20]: Oh, that is worse than I thought.

Comment (KrzysztofCwalina) [reply sdk/core/Azure.Core/api/Azure.Core.net461.cs:20]: I think two enums, but one enum for this method and one different for the other solves all problems that people had.

**Pull request closed without merging**